File: Java_Backend/src/main/java/com/netlink/rsk/model/TrainerAttendanceVerification.java
package com.netlink.rsk.model;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;

@Data
@Entity
@Table(name = "Trainerattendanceverification",catalog = "Training")
public class TrainerAttendanceVerification implements Serializable {

    @Id
    @Column(name = "Trainerverificationid", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long trainerVerificationId;

    @OneToOne
    @JoinColumn(name="Trainerattendanceid",referencedColumnName = "Trainerattendanceid")
    private TrainerAttendance trainerAttendance;

    @OneToOne
    @JoinColumn(name="Trainerid",referencedColumnName = "Trainerid")
    private TrainerMaster trainer;

    @OneToOne
    @JoinColumn(name="Batchid",referencedColumnName = "Batchid")
    private BatchMaster batch;

    @Column(name="Verify")
    private Boolean verify;

    @Column(name="Verificationremark")
    private String verificationRemark;

    @Column(name="Rating")
    private Integer rating;

    @Column(name="Ratingremark")
    private String  ratingRemark;

    @Column(name = "Createdby")
    private Integer createdBy;

    @Column(name = "Createdon")
    private Date createdOn;

    @OneToOne
    @JoinColumn(name = "Updatedby", referencedColumnName = "Userid")
    private UserMaster updatedBy;

    @Column(name = "Updatedon")
    private Date updatedOn;

}

File: client/src/app/layouts/training/model/trainer-attendance-verification.ts
import { User } from "src/app/views/pages/login/model/user.model";
import { BatchMaster } from "./batch-master";
import { TrainerAttendance } from "./trainer-attendance";
import { TrainerMaster } from "./trainer-master";

export class TrainerAttendanceVerification {
    trainerVerificationId: number;
    trainerAttendance: TrainerAttendance;
    verify: boolean;
    verificationRemark: string;
    rating: number;
    ratingRemark: string;
    createdBy: number;
    createdOn: Date;
    updatedBy: User;
    updatedOn: Date;
    trainer:TrainerMaster;
    batch:BatchMaster;
}
